lib/detail_page.dart

```diff
@@ -39,7 +41,11 @@ class _DetailPageState extends State<DetailPage> {
   @override
   Widget build(BuildContext context) {
     final deviceModel = context.watch<DeviceModel>();
+    final store = context.watch<DeviceStore>();
     final navigator = Navigator.of(context);
+    final release = deviceModel.releases
+        ?.singleWhereOrNull((r) => r.version == store.selectedReleaseVersion);
```

Comment:
whenever possible, we could hide even this kind of simpler logic in the view model using helper methods and properties. it's not much of a difference but helps to keep the widget tree as dumb/logic-free as possible. for example, something like

```dart
final release = deviceModel.findRelease(store.selectedReleaseVersion);
```

Reply:
:+1: